File: src/main/java/io/github/hdzitao/editstarters/ohub/metadata/OHubMetaData.java
package io.github.hdzitao.editstarters.ohub.metadata;

import io.github.hdzitao.editstarters.version.Version;
import io.github.hdzitao.editstarters.version.Versions;
import org.apache.commons.lang3.StringUtils;

/**
 * oHub配置元素
 *
 * @version 3.2.0
 */
public class OHubMetaData {
    private String versionRange;
    private String metadataConfig;
    private boolean enable = false; // 默认关闭

    public String getVersionRange() {
        return versionRange;
    }

    public void setVersionRange(String versionRange) {
        this.versionRange = versionRange;
    }

    public String getMetadataConfig() {
        return metadataConfig;
    }

    public void setMetadataConfig(String metadataConfig) {
        this.metadataConfig = metadataConfig;
    }

    public boolean isEnable() {
        return enable;
    }

    public void setEnable(boolean enable) {
        this.enable = enable;
    }

    /**
     * 检查版本
     */
    public boolean checkVersionRange(Version version) {
        if (StringUtils.isEmpty(versionRange)) {
            return false;
        }

        return Versions.parseRange(versionRange).match(version);
    }
}

File: src/main/kotlin/io/github/hdzitao/editstarters/ohub/metadata/OHubMetaData.kt
package io.github.hdzitao.editstarters.ohub.metadata

import io.github.hdzitao.editstarters.version.Version
import io.github.hdzitao.editstarters.version.Versions

/**
 * oHub配置元素
 *
 * @version 3.2.0
 */
class OHubMetaData(
    var versionRange: String? = null,
    var metadataConfig: String? = null,
    var enable: Boolean = false // 默认关闭
) {

    /**
     * 检查版本
     */
    fun checkVersionRange(version: Version): Boolean {
        return if (versionRange.isNullOrBlank()) {
            false
        } else {
            Versions.parseRange(versionRange!!).match(version)
        }
    }
}
